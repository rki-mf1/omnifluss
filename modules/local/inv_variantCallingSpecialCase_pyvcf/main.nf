process RESCUE_VARIANTS {
    tag "$meta.id"

    input:
    tuple val(meta), path(vcf)

    output:
    tuple val(meta), path('*.special_case_variant_mask.bed'), emit: bed
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.0.1'
    """
    python $projectDir/bin/rescue_HQ_variants.py $vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rescue_HQ_variants.py: $VERSION
    END_VERSIONS
    """
    
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.0.1'
    """
    touch "${prefix}.special_case_variant_mask.bed"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rescue_HQ_variants.py: $VERSION
    END_VERSIONS
    """
}
