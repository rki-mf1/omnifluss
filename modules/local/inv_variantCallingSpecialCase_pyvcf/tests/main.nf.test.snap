{
    "sarscov2 - vcf": {
        "content": [
            [
                [
                    {
                        "id": "test2"
                    },
                    "test2.special_case_variant_mask.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-01T13:47:48.361394427"
    },
    "influenza - vcf": {
        "content": [
            [
                [
                    {
                        "id": "test.spec_case.filtered"
                    },
                    "test.special_case_variant_mask.bed:md5,bdd89429997e007414772c250cd704de"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-01T13:47:38.759932808"
    },
    "influenza - vcf - stub": {
        "content": [
            [
                [
                    {
                        "id": "test.spec_case.filtered"
                    },
                    "test.spec_case.filtered.special_case_variant_mask.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-01T13:47:43.216832968"
    }
}